### ISSUE TITLE
[Settings] emphasize common URLs

### ISSUE THREAD

**Issue by MenaiAla** [forem/forem#9287]
### Category : 
User Experience
### Description :
In the settings page, you provide long-form of URLs ( 11 URLs once time ) which makes the scrolling too long. Most users have just 2 or 3 common URLs

## Suggestion

Provide just the popular URLs and let users expand the UI to provide other URLs.

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by Veggier** [comment]
I will try to work on this issue

**Comment by rhymes** [comment]
Hi @Veggier, this feature hasn't been approved yet :) - see https://github.com/forem/forem/issues/9287#issuecomment-657557578 for an explanation - I think we should additionally wait for the designers to weigh in as this pertains to a larger conversation on how the user's profile should look like.

Thanks for offering though, we can pick it up later on when we've reached a consensus

**Comment by Veggier** [comment]
ok

**Comment by rhymes** [comment]
The `Links` section should be reduced to a list of input fields where you can input any link. Then auto-detect if it is one of the "known" options and display a favicon for it. If you are worried about spam accounts putting spam links there, then just have a list of known-good domains, and others can be "pending approval". But this would fix the following issues:

* <img alt="Glitch icon" src='https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/glitch.svg' height='24'> [#7697 - Add Glitch link](https://github.com/forem/forem/issues/7697) 
* <img alt="DeviantArt icon" src='https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/deviantart.svg' height='24'> [#6117 - Add DeviantArt link](https://github.com/forem/forem/issues/6117)
* <img alt="Pinterest icon" src='https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/pinterest.svg' height='24'> [#6118 - Add Pinterest link](https://github.com/forem/forem/issues/6118)
* <img alt="CodePen icon" src='https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/codepen.svg' height='24'> [#6087 - Add CodePen link](https://github.com/forem/forem/issues/6087)
* <img alt="Twitter icon" src='https://cdn.jsdelivr.net/npm/simple-icons@3.0.1/icons/twitter.svg' height='24'> [#9646 - Add Twitter link](https://github.com/forem/forem/issues/9646)

With this approach the length of the page is reduced, since we only need the link fields for links the user added + 1 empty field for them to add the next link to. Also, people are not arbitrarily prevented from linking to whatever sites they want. Any additional requests in the future would only require vetting a domain name then adding it to a list, and a new favicon. There are plenty of libraries, like [simple-icons](https://simpleicons.org) that can dynamically handle icons via CDN. You could even add to the white-list all of the social sites that simple-icons supports currently preemptively, thus making the process much smoother (one-time vetting process, followed by much more rare requests).

_Originally posted by @TheJaredWilcurt in https://github.com/forem/forem/issues/9220#issuecomment-669632071_

**Comment by ludwiczakpawel** [comment]
Here's an idea of how the UI/UX could look like:

![image](https://user-images.githubusercontent.com/108287/89522842-29ace980-d7e2-11ea-84d0-441acbaa6186.png)

And also bunch of icons I could ship like right away (since as of today, we prefer to store icons in our repo):

![image](https://user-images.githubusercontent.com/108287/89523362-ff0f6080-d7e2-11ea-8573-88a440b28204.png)

**Comment by rhymes** [comment]
@citizen428 @Ridhwana is this still relevant in the context of the work you're doing about profile generalization?

**Comment by Ridhwana** [comment]
@rhymes I don't think so, the underlying context has changed since the admin will now decide what links they require.

**Comment by rhymes** [comment]
Thanks, closing this then

**Comment by citizen428** [comment]
Actually, I think we had discussed on Slack with @ludwiczakpawel that it would still be nice to at least have automatic favicons for popular websites. But I guess we can open a new issue for this in the future if we really decide to go ahead with the idea.

**Comment by ludwiczakpawel** [comment]
I don't think there's anything wrong with this issue, even with our updated implementation of the profile fields. I think this is still valid - not only issue but also a potential solution. But I leave the decision to others - I know @Ridhwana had slightly different point of view on this.

**Comment by Ridhwana** [comment]
I think we definitely should implement this in the future. My discussion with Pawel on Thursday was that we provided a solution to the original issue created ( through the first iteration of profile generalization by giving the admin more control over how many link fields they expose for their Forem ) and so it would make sense to close this issue and open a new issue to tackle a second iteration of this, but like I mentioned @ludwiczakpawel like I mentioned in our discussion,  I don't feel strongly about this so if you'd like the next phase to form as part of this issue, you can reopen it 🙂
